ee/api/test/test_organization.py

```diff
@@ -274,3 +274,20 @@ def test_feature_available_self_hosted_license_expired(self, patch_post):
             self.organization.refresh_from_db()
             self.assertFalse(self.organization.is_feature_available("whatever"))
         License.PLANS = current_plans
+
+    def test_get_organization_restricted_teams_hidden(self):
+        self.organization_membership.level = OrganizationMembership.Level.MEMBER
+        self.organization_membership.save()
+        Team.objects.create(
+            organization=self.organization,
+            name="FORBIDDEN",
+            access_control=True,
```

Comment:
I guess this is the line that forbids access?

Would be good to add a third team here where someone does have access so you can see both sides of the permissions setup